src/lib.rs

```diff
@@ -342,14 +341,10 @@ pub(crate) unsafe fn rav1d_open(c_out: &mut *mut Rav1dContext, s: &Rav1dSettings
         (*c).task_thread.cond_signaled = AtomicI32::new(0);
         (*c).task_thread.inited = 1 as c_int;
     }
-    if (*c).n_fc > 1 as c_uint {
-        (*c).frame_thread.out_delayed = calloc(
-            (*c).n_fc as usize,
-            ::core::mem::size_of::<Rav1dThreadPicture>(),
-        ) as *mut Rav1dThreadPicture;
-        if ((*c).frame_thread.out_delayed).is_null() {
-            return error(c, c_out, &mut thread_attr);
-        }
+    if (*c).n_fc > 1 {
+        (*c).frame_thread.out_delayed = std::iter::repeat_with(Rav1dThreadPicture::default)
+            .take((*c).n_fc as usize)
+            .collect();
     }
```

Comment:
`c: *mut Rav1dContext` is currently zero-initialized (see the `memset` above in `fn rav1d_open`).  So either `out_delayed` needs to be valid for zeroed bits, or we need to initialize it explicitly in an `else` case as well.  And until we remove the `memset` and initialize things directly, we should use `addr_of_mut!((*c).frame_thread.out_delayed).write(...)` to write to `out_delayed` to avoid dropping the zero-initialized `out_delayed` if that's not a valid representation.  Thus, I'd recommend something like this:

```suggestion
    addr_of_mut!((*c).frame_thread.out_delayed).write(if (*c).n_fc > 1 {
        (0..(*c).n_fc).map(|_| Default::default()).collect()
    } else {
        Box::new([])
    });
```

Reply:
makes sense! fixed